packages/contracts/contracts/appDefinitions/HighRollerApp.sol

```diff
@@ -0,0 +1,141 @@
+pragma solidity 0.4.25;
+pragma experimental "ABIEncoderV2";
+
+import "../lib/Transfer.sol";
+
+
+contract HighRollerApp {
+
+  enum ActionType {
+    COMMIT_TO_HASH,
+    COMMIT_TO_NUM,
+    REVEAL
+  }
+
+  enum Stage {
+    COMMITING_HASH, COMMITTING_NUM, REVEALING, DONE
+  }
+
+  enum Player {
+    FIRST,
+    SECOND,
+    TIE // TODO How should we handle this? Doesn't make sense that it is a "Player"
+  }
+
+  struct AppState {
+    address[2] playerAddrs;
+    Stage stage;
+    bytes32 commitHash;
+    uint256 commitNum;
+    uint256[2] playerFirstRoll; // TODO Does it make sense to store these in AppState?
```

Comment:
Probably don't need it in the appstate

Reply:
How would `applyAction` return the roll then?
I can also add a new pure function to return the rolls after the game is DONE.
Another option is for `applyAction` to return the keccak(s1*s2) hash and the client will figure out the rolls since it is just for the UI. Might also be more efficient than running it on the blockchain no?

Reply:
You just need to compute the winner one time, you don't need to store the rolls. `resolve` should determine who won based on the splitting of the random number.

Reply:
The UI can be indicative of the potential outcome, not on the state. 

There exists a function that maps both secrets and the player addresses to a winner or loser and it can exist clientside.